batch/batch/utils.py

```diff
@@ -161,14 +176,36 @@ async def query_billing_projects(db, user=None, billing_project=None):
 LOCK IN SHARE MODE;
 '''
 
-    def record_to_dict(record):
-        if record['users'] is None:
-            record['users'] = []
-        else:
-            record['users'] = json.loads(record['users'])
-        return record
+    billing_projects = [users_record_to_dict(record) async for record in db.execute_and_fetchall(sql, tuple(args))]
```

Comment:
This isn't new in this PR so feel free to take or leave this critique, but I personally find functions that both mutate inputs hard to reason about, as I would expect a void function to mutate and a function with a return value to leave its inputs alone. I might prefer to write this the following way:

```python
billing_projects = []
async for record in db.execute_and_fetchall(sql, tuple(args)):
    record['users'] = json.loads(record['users']) if record['users'] is not None else []
    billing_projects.append(record)    
```